### PR TITLE
Renaming of API functions; documentation pass

### PR DESCRIPTION
**Pull request by rich-iannone**
The changes in this PR are focused on changing names of some helper functions, improving the internal help files, updating the pkgdown site, and updating the README.

Fixes: https://github.com/rstudio/gt/issues/426
Fixes: https://github.com/rstudio/gt/issues/421
Fixes: https://github.com/rstudio/gt/issues/322
Fixes: https://github.com/rstudio/gt/issues/321

### CODE REVIEW

File: R/text_transform.R [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-#'   footnote text. These helper functions are: [cells_title()],
-#'   [cells_column_labels()], [cells_group()], [cells_stub()], [cells_data()],
-#'   and [cells_summary()]. Please see the help article \link{location_cells}
-#'   for more information on how these helper functions can be used.
-#'   Additionally, we can enclose several `cells_*()` calls within a `list()` if
-#'   we wish to link the footnote text to different types of locations (e.g.,
-#'   cell data values, stub group headings, the table title, etc.).
+#'   transformation. Only the [cells_body()] helper function can be used here.
+#'   We can enclose several [cells_body()] calls within a `list()` if we wish to
+#'   make the transformation happen at different body cell locations.
```
Comment (jcheng5) [inline R/text_transform.R]: Actually other cell types work as well

File: README.Rmd [hunk 2]
```diff
@@ -21,17 +21,17 @@ knitr::opts_chunk$set(
...
+With the **gt** package, anyone can make wonderful-looking tables using the **R** programming language. The **gt** philosophy: we can construct a wide variety of useful tables with a cohesive set of table parts. These include the *table header*, the *stub*, the *column labels* and *spanner column labels*, the *table body*, and the *table footer*.
 
 <p align="center"><img src="man/figures/gt_parts_of_a_table.svg" width="800px"></p>
 
-It all begins with preprocessed **table data** (be it a tibble or a data frame). You decide how to compose your **gt table** with the elements you need for the task at hand. Output can either be in the form of **HTML**, **LaTeX**, or **RTF**. All work beautifully inside **R Markdown** documents.
+It all begins with preprocessed **table data** (be it a tibble or a data frame). You decide how to compose your **gt table** with the elements you need for the task at hand.
```
Comment (jcheng5) [inline README.Rmd]:
```suggestion
It all begins with preprocessed **table data** (be it a tibble or a data frame). You decide how to compose your **gt table** with the elements you need for the task at hand. Currently, **gt** supports **HTML** output, with **LaTeX** and **RTF** planned for the future.
```

File: R/text_transform.R (hunk 1 above)
Comment (rich-iannone) [reply R/text_transform.R]: Now changed, thanks!

File: README.Rmd (hunk 2 above)
Comment (rich-iannone) [reply README.Rmd]: I made the change, reads much better now.

File: R/helpers.R [hunk 3]
```diff
@@ -532,6 +532,26 @@ cells_row_groups <- function(groups = TRUE) {
...
 
+#' @noRd
+#' @export
+cells_group <- function(groups = TRUE) {
+
+  warning("The `cells_group()` function is deprecated and will soon be removed\n",
+          " * Use the `cells_row_groups()` function instead",
+          call. = FALSE)
+
+  # Capture expression for the `groups` argument
```
Comment (jcheng5) [inline R/helpers.R]: Could the body of this function just be the deprecation warning, and then `cells_row_groups({{groups}})`?

File: R/helpers.R [hunk 4]
```diff
@@ -708,6 +728,32 @@ cells_body <- function(columns = TRUE,
...
+#' @noRd
+#' @export
+cells_data <- function(columns = TRUE,
+                       rows = TRUE) {
+
+  warning("The `cells_data()` function is deprecated and will soon be removed\n",
+          " * Use the `cells_body()` function instead",
+          call. = FALSE)
+
+  # Capture expressions for the `columns` and `rows` arguments
```
Comment (jcheng5) [inline R/helpers.R]: Same here, could it just be the warning and then `cells_data({{columns}}, {{rows}})`?

File: R/helpers.R (hunk 3 above)
Comment (rich-iannone) [reply R/helpers.R]: This was the first thing I tried but it resulted in an error (probably due to the unquoting). I did use `!!` on the arg value but it still didn’t work out.

File: R/helpers.R (hunk 4 above)
Comment (rich-iannone) [reply R/helpers.R]: I tried this with `!!` but not with the double curly brace operator. Let me see if that makes a difference.

Comment (rich-iannone) [reply R/helpers.R]: Whoa. With the `{{ }}` we get an infinite loop!

Comment (rich-iannone) [reply R/helpers.R]: I think we may just have to keep the exact code in both places. The good thing is that I don't expect `cells_body()` or `cell_row_groups()` to change in ~6 months time (when I might change the `warning()` to a `stop()`).

**Pull request merged**